wab/src/main/java/org/apache/unomi/web/ContextServlet.java

```diff
@@ -136,11 +136,17 @@ public void service(ServletRequest request, ServletResponse response) throws Ser
             try {
                 contextRequest = mapper.readValue(factory.createParser(stringPayload), ContextRequest.class);
             } catch (Exception e) {
+                ((HttpServletResponse)response).sendError(HttpServletResponse.SC_BAD_REQUEST, "Check logs for more details");
                 logger.error("Cannot read payload " + stringPayload, e);
                 return;
             }
             scope = contextRequest.getSource().getScope();
         }
+        if (httpMethod.equals("POST")){
+            ((HttpServletResponse)response).sendError(HttpServletResponse.SC_BAD_REQUEST, "Check logs for more details");
+            logger.error("Invalid Resource, please check the request");
+            return;
+        }
```

Comment:
So you want to reject only POST requests? Perhaps instead of defining which requests we don't support we should define which ones we support and throw the error otherwise? 

Also check the above treatment of OPTIONS http method above - isn't it a bit too late where we check it, shouldn't we group the checks together and use similar code?

Reply:
see above comment. We are not changing the handling of the different requests, simply making sure errors that are logged as errors also return an HTTP error code.

Reply:
Yeah I understand the intention, but above code suggests otherwise - it sends an error when HTTP method is POST if I see correctly. I just wanted to confirm that is what we want. Perhaps this if condition should be in the catch block or something?

Also if that's what we want, I would move this if block a bit higher, close to the if block that checks for OPTIONS httpMethod (this is more question of taste).

Reply:
I am also saying that if we really want to block POST method, then we should probably also block DELETE, PATCH, PUT, etc., so it would be better to invert the condition and block anything else than GET and OPTIONS (whitelist instead of blacklist).